code/game/turfs/open.dm

```diff
@@ -343,6 +343,9 @@
 			if(L.stat == DEAD)
 				continue
 
+			if(L.moving_diagonally == FIRST_DIAG_STEP)
+				continue
```

Comment:
prevents lava from burning when you clearly didn't step on lava.

Can maybe be atomised if you want, but this caused xeno to burn themselfs on magmoor